src/wovnio/wovnphp/Headers.php

```diff
@@ -346,4 +346,19 @@ public function shouldRedirect()
 
         return $cookieLang && ($requestLang !== $cookieLang) && ($requestLang === $this->store->defaultLang());
     }
+
+    public function getCanonicalUrl()
+    {
+        $canonicalUrl = $this->protocol . '://' . $this->host . $this->pathnameKeepTrailingSlash;
+        if ($this->store->settings['query_canonical_significance'])
+        {
+            $canonicalUrl = $canonicalUrl . '?' . $this->query;
+        }
+        return $canonicalUrl;
+    }
+
+    public function isGoogleBot()
```

Comment:
just an early review, but don't name things specific to google bot. You can bet we're going to add more search engines later 😉

Reply:
We are not gonna try to support Yahoo, yandex or baidu right.

right?